sycl/doc/extensions/supported/sycl_ext_intel_esimd/sycl_ext_intel_esimd_functions.md

```diff
@@ -0,0 +1,647 @@
+/*lacc-4*/ void scatter(AccessorT acc, OffsetSimdViewT byte_offsets, simd<T, N> vals,
+                        PropertyListT props = {});
+
+// scatter to shared local memory.
+template <typename T, int N, int VS = 1, typename PropertyListT = empty_properties_t>
+/*slm-1*/ void scatter(simd<uint32_t, N / VS> byte_offsets, simd<T, N> vals,
+                       simd_mask<N / VS> mask, PropertyListT props = {});
+/*slm-2*/ void scatter(simd<uint32_t, N / VS> byte_offsets,
+                       simd<T, N> vals, PropertyListT props = {});
+
+// scatter to shared local memory  - similar to (slm-1,2), but the `byte_offsets` is `simd_view`.
+template <typename T, int N, int VS = 1, typename OffsetSimdViewT, typename PropertyListT = empty_properties_t>
+/*slm-3*/ void scatter(OffsetSimdViewT byte_offsets, simd<T, N> vals,
+                       simd_mask<N / VS> mask, PropertyListT props = {});
+/*slm-4*/ void scatter(OffsetSimdViewT byte_offsets, simd<T, N> vals,
+                       PropertyListT props = {});
+```
+
+### Description
+`(usm-*)`: Stores ("scatters") the vector `vals` to global memory locations addressed by the base USM pointer `p` and byte-offsets `byte_offsets`.  
+`(acc-*)`, `(lacc-*)`: Stores ("scatters") the vector `vals` to memory locations addressed by the the accessor and byte-offsets `byte_offsets`.  
+`(slm-*)`: Stores ("scatters") the vector `vals` to shared local memory locations addressed by `byte_offsets`.  
+The parameter `byte_offsets` has is a vector of integer 64-bit type elements for `(usm-*)`, 32-bit integer elements for `(lacc-*)` and `(slm-*)`, 32-bit integer elements for `(acc-*)` in [stateful](#statelessstateful-memory-mode) mode, and 64-bit integer elements for `(acc-*)` in [stateless](#statelessstateful-memory-mode) mode.  
+The optional parameter `pred` provides a `simd_mask`. If some element in `pred` is zero, then the store to the corresponding memory location is skipped.  
+The optional [compile-time properties](#compile-time-properties) list `props` may specify `alignment` and/or `cache-hints`. The cache-hints are ignored for `(lacc-*)` and `(slm-*)` functions.
+
+## atomic_update(...)
+
+### atomic_update() with 0 operands (inc, dec, load)
+```C++
+// Atomic update the USM memory locations - zero operands (dec, load, etc.).
+template <atomic_op Op, typename T, int N, typename Toffset, typename PropertyListT = empty_properties_t>
+/*usm-0op-1*/ simd<T, N> atomic_update(T *p, simd<Toffset, N> byte_offset, simd_mask<N> mask, props = {});
+/*usm-0op-2*/ simd<T, N> atomic_update(T *p, simd<Toffset, N> byte_offset,props = {});
+
+// Similar to (usm-0op-1,2), but `byte_offset` is `simd_view`.
+template <atomic_op Op, typename T, int N, typename OffsetObjT, typename RegionT,
+          typename PropertyListT = detail::empty_properties_t>
+/*usm-0op-3*/ simd<T, N> atomic_update(T *p, simd_view<OffsetObjT, RegionT> byte_offset, simd_mask<N> mask, props = {});
```

Comment:
Should these be OffsetSimdViewT?

Reply:
That is how the actual (usm-au0-3) is defined now. I did not plan changing the func definition inside memory.hpp in this patch. This one is NFC/DOC (maybe fix comment in memory.hpp at most).

Reply:
My reply is the same as [here](https://github.com/intel/llvm/pull/13071#discussion_r1535871779)